utils/src/main/scala/za/co/absa/enceladus/utils/testUtils/LoggerTestBase.scala

```diff
@@ -21,16 +21,36 @@ import org.apache.spark.sql.DataFrame
 import org.slf4j.{Logger, LoggerFactory}
 
 trait LoggerTestBase {
+  import LoggerTestBase.LogLevel._
+
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
-  protected def logDataFrameContent(df: DataFrame): Unit = {
-    logger.debug(df.schema.treeString)
+  def logLevelToLogFunction(logLevel: LogLevel): String => Unit = {
+    logLevel match {
+      case TRACE => logger.trace
+      case DEBUG => logger.debug
+      case INFO  => logger.info
+      case WARN  => logger.warn
+      case _     => logger.error
+    }
+  }
+
+  protected def logDataFrameContent(df: DataFrame, logLevel: LogLevel = DEBUG): Unit = {
```

Comment:
Consider having a function that gets a string out of a dataframe, say, `getDataFrameShowString(df)`.

Then, you won't need `logLevelToLogFunction()` and just use:
```scala
log.error(getDataFrameShowString(df))
```

Also, `getDataFrameShowString()` is easily testable, while `logDataFrameContent()` is no so easily testable.

Reply:
The testability is a point. On the other hand this is just a logging function, not really needs testing.
Also the point the function was created, was to just throw in the dataset and forget about it. The "string extracting" function would be a two step solution again.

Reply:
Testability is not the only point.
* Current implementation required you to create a sealed trait with all the case objects that replicate log levels of the underlying logging library. Looks a little bit redundant.
* Having contents of a DataFrame as a string is more reusable. For instance, if you want to `assert` differences of outputs returned by `.show()` you can use the same function for converting DataFrame contents to String.
* Fetching contents of a DataFrame and logging seems like 2 orthogonal responsibilities. When split they can be reused better.

Reply:
OK I created the function(s) to get the `DataFrame` data as string. But @yruslan I dare you I expect you to use it in at least one of your next 3 pull requests. 😉